### PR TITLE
media-video/shotcut: version bump to 20.10.31

### PR DESCRIPTION
**Pull request by xdch47**
Closes: https://bugs.gentoo.org/688850
Bug: https://bugs.gentoo.org/744199
Package-Manager: Portage-3.0.9, Repoman-3.0.2
Signed-off-by: Felix Neumärker <xdch47@posteo.de>

### CODE REVIEW

**Review by juippis — COMMENTED** [review]
Looks good, but please split adding and removing ebuilds into separate commits. So this PR should have 2 commits instead of one.

File: media-video/shotcut/shotcut-20.10.31.ebuild [hunk 1]
```diff
@@ -3,59 +3,47 @@
...
+COMMON_DEPEND="dev-qt/qtcore:5
 	dev-qt/qtdeclarative:5[widgets]
 	dev-qt/qtgui:5
 	dev-qt/qtmultimedia:5
 	dev-qt/qtnetwork:5
 	dev-qt/qtopengl:5
 	dev-qt/qtprintsupport:5
 	dev-qt/qtsql:5
-	dev-qt/qtwebkit:5
+	dev-qt/qtwebengine:5
```
Comment (a17r) [inline media-video/shotcut/shotcut-20.10.31.ebuild:26]: Where is that dependency coming from?

File: media-video/shotcut/shotcut-20.10.31.ebuild [hunk 2]
```diff
@@ -3,59 +3,47 @@
 
 EAPI=7
 
-inherit flag-o-matic qmake-utils xdg
+inherit qmake-utils xdg
 
 DESCRIPTION="A free, open source, cross-platform video editor"
-HOMEPAGE="https://www.shotcut.org/ https://github.com/mltframework/shotcut/"
+HOMEPAGE="https://www.shotcut.org https://github.com/mltframework/shotcut"
```
Comment (a17r) [inline media-video/shotcut/shotcut-20.10.31.ebuild:9]: Keep trailing slash on `https://www.shotcut.org/`.

File: media-video/shotcut/shotcut-20.10.31.ebuild [hunk 3]
```diff
@@ -3,59 +3,47 @@
...
 
-BDEPEND="
-	dev-qt/linguist-tools:5
-"
-COMMON_DEPEND="
-	dev-qt/qtcore:5
+BDEPEND="dev-qt/linguist-tools:5"
+
+COMMON_DEPEND="dev-qt/qtcore:5
 	dev-qt/qtdeclarative:5[widgets]
```
Comment (a17r) [inline media-video/shotcut/shotcut-20.10.31.ebuild:19]: Please keep the existing ebuild style and avoid unnecessary changes, same below.

File: media-video/shotcut/shotcut-20.10.31.ebuild [hunk 4]
```diff
@@ -3,59 +3,47 @@
...
 	dev-qt/qtgraphicaleffects:5
 	dev-qt/qtquickcontrols:5
-	virtual/jack
-"
-
-src_prepare() {
-	default
-
-	sed -i -e '/QT.*private/d' \
-		src/src.pro || die
```
Comment (a17r) [inline media-video/shotcut/shotcut-20.10.31.ebuild:50]: This is no longer necessary?

**Review by a17r — CHANGES_REQUESTED** [review]

File: media-video/shotcut/shotcut-20.10.31.ebuild [hunk 5]
```diff
@@ -3,59 +3,47 @@
...
-	dev-qt/qtwebkit:5
+	dev-qt/qtwebengine:5
 	dev-qt/qtwebsockets:5
 	dev-qt/qtwidgets:5
 	dev-qt/qtxml:5
-	>=media-libs/mlt-6.18.0[ffmpeg,frei0r,jack,qt5,sdl,xml]
-	media-libs/webvfx
-	media-video/ffmpeg
-"
+	>=media-libs/mlt-6.22.1[ffmpeg,frei0r,jack,melt,qt5,sdl,xml]
```
Comment (a17r) [inline media-video/shotcut/shotcut-20.10.31.ebuild:30]: Please make this `melt(+)` in preparation of `media-libs/mlt` losing a nonsensical IUSE.

**Review by a17r — CHANGES_REQUESTED** [review]

**Comment by a17r** [comment]
This is not your first contribution. Would you like to become new maintainer of this package?

File: media-video/shotcut/shotcut-20.10.31.ebuild (hunk 4 above)
Comment (xdch47) [reply media-video/shotcut/shotcut-20.10.31.ebuild:50]: It builds fine without in my case - The line which got removed is:
```QT       += qml-private core-private quick-private gui-private``` (Using ```/usr/lib64/qt5/mkspecs/modules/qt_lib_{qml,core,quick,gui}_private.pri```)
Don't know what has been the problem - maybe thoses mkspecs or headers were missing in earlier QT version?

**Comment by xdch47** [comment]
> This is not your first contribution. Would you like to become new maintainer of this package?

Yes, I would like to - changed `metadata.xml` as well. 
--> https://bugs.gentoo.org/687148 has to be reopened then, right?

File: media-video/shotcut/shotcut-20.10.31.ebuild (hunk 1 above)
Comment (xdch47) [reply media-video/shotcut/shotcut-20.10.31.ebuild]: Not neccessary - sorry.

File: media-video/shotcut/shotcut-20.10.31.ebuild (hunk 4 above)
Comment (a17r) [reply media-video/shotcut/shotcut-20.10.31.ebuild:50]: Traced it back to this PR comment: https://github.com/gentoo/gentoo/pull/14393#issuecomment-578481918

Basically silencing a bogus warning.
